state/factory/factory.go

```diff
@@ -83,10 +95,10 @@ func PrecreatedTrieDBOption(kv db.KVStore) Option {
 		}
 		sf.dao = kv
 		// get state trie root
-		if sf.rootHash, err = sf.getRoot(trie.AccountKVNameSpace, AccountTrieRootKey); err != nil {
+		if sf.rootHash, err = sf.getRoot(AccountKVNameSpace, AccountTrieRootKey); err != nil {
 			return errors.Wrap(err, "failed to get accountTrie's root hash from underlying DB")
 		}
-		if sf.accountTrie, err = trie.NewTrie(sf.dao, trie.AccountKVNameSpace, sf.rootHash); err != nil {
+		if sf.accountTrie, err = trie.NewTrie(sf.dao, AccountKVNameSpace, trie.RootHashOption(sf.rootHash)); err != nil {
 			return errors.Wrap(err, "failed to generate accountTrie from config")
```

Comment:
we should add KeyLengthOption(PKHashSize) to enforce all keys are 20-bytes in length?

Reply:
20 is the default length